node/src/neighborhood/mod.rs

```diff
@@ -4235,28 +4352,66 @@ mod tests {
 
     #[test]
-    fn handle_new_public_ip_changes_public_ip_and_nothing_else() {
+    fn handle_new_public_ip_changes_public_ip_and_announces_the_change() {
         init_test_logging();
         let subject_node = make_global_cryptde_node_record(1234, true);
-        let neighbor = make_node_record(1050, true);
-        let mut subject: Neighborhood = neighborhood_from_nodes(&subject_node, Some(&neighbor));
         let new_public_ip = IpAddr::from_str("4.3.2.1").unwrap();
+        let (hopper, _, hopper_recording_arc) = make_recorder();
+        let (subject, one_neighbor, another_neighbor) = {
+            let mut subject: Neighborhood = neighborhood_from_nodes(&subject_node,
+                Some(&make_node_record(7654, true)));
+            subject.gossip_producer_opt = Some(Box::new (GossipProducerReal::new()));
+            subject.hopper_no_lookup_opt = Some (hopper.start().recipient());
+            let db = &mut subject.neighborhood_database;
+            let one_neighbor = db.add_node(make_node_record(1234, true)).unwrap();
+            let another_neighbor = db.add_node(make_node_record(2345, true)).unwrap();
+            let originate_only_neighbor = db.add_node(make_node_record(3456, false)).unwrap();
+            let not_a_neighbor = db.add_node(make_node_record(4567, false)).unwrap();
+            db.add_arbitrary_full_neighbor(subject_node.public_key(), &one_neighbor);
+            db.add_arbitrary_full_neighbor(subject_node.public_key(), &another_neighbor);
+            db.add_arbitrary_full_neighbor(subject_node.public_key(), &originate_only_neighbor);
+            db.add_arbitrary_full_neighbor(&not_a_neighbor, &another_neighbor);
+            let system = System::new("test");
+
+            subject.handle_new_public_ip(NewPublicIp {
+                new_ip: new_public_ip,
+            });
 
-        subject.handle_new_public_ip(NewPublicIp {
-            new_ip: new_public_ip,
-        });
-
+            System::current().stop();
+            system.run();
+            (subject, one_neighbor, another_neighbor)
+        };
+        let db = &subject.neighborhood_database;
         assert_eq!(
-            subject
-                .neighborhood_database
-                .root()
+            db.root()
                 .node_addr_opt()
                 .unwrap()
                 .ip_addr(),
             new_public_ip
         );
         TestLogHandler::new()
             .exists_log_containing("INFO: Neighborhood: Changed public IP from 1.2.3.4 to 4.3.2.1");
+        let hopper_recording = hopper_recording_arc.lock().unwrap();
+        assert_eq! (hopper_recording.len(), 2);
+        let check_hopper_message = |idx: usize, expected_key: &PublicKey| {
+            let pkg: &NoLookupIncipientCoresPackage = hopper_recording.get_record(idx);
+            assert_eq! (&pkg.public_key, expected_key);
+            assert_eq! (&pkg.node_addr, db.node_by_key(expected_key).as_ref().unwrap().metadata.node_addr_opt.as_ref().unwrap());
```

Comment:
I'd rather see the expected part be a variable defined at the line before so that we haven't got such a long chain of function calls within the assertion macro.

Reply:
I think it could have worked with just a couple of additional line breaks, but I put in a temporary variable for you.